scalalib/src/mill/scalalib/CoverageModule.scala

```diff
@@ -0,0 +1,309 @@
+package mill.scalalib
+
+import scoverage.{Constants => ScoverageConstants};
+
+import mill.Cross
+import mill.scalalib.{SbtModule, PublishModule, Dep, CrossSbtModule, DepSyntax}
+import mill.scalalib.publish.{PomSettings, License, Developer, SCM}
+import mill.define.TaskModule
+import scala.sys.process.Process
+import mill._, scalalib._
+import mill.util.Logger
+import ammonite.ops._
+import java.io.File
+
+import scoverage.IOUtils;
+import scoverage.Serializer;
+import scoverage.Coverage;
+import scoverage.report.{
+  CoverageAggregator,
+  CoberturaXmlWriter,
+  ScoverageHtmlWriter,
+  ScoverageXmlWriter
+}
+
+/** A little trait to mix in within a module to add coverage.
+  *
+  * {{{
+  * object coverage extends Tests with CoverageModule {
+  *     def testIvyDeps = Agg(ivy"org.scalatest::scalatest:3.0.1")
+  *     def testFrameworks = Seq("org.scalatest.tools.Framework")
+  * }
+  * }}}
```

Comment:
I'd like to figure out how to add `trait Coverage extends CoverageModule with Tests` to `ScalaModule`.

Reply:
No idea what this means